tools/test-setup.sh

```diff
@@ -302,6 +302,10 @@ if [[ -n "${CI:-}" ]]; then
         log warning "Removing unexpected ~/.ansible folder found on CI to avoid test contamination."
         rm -rf "${HOME}/.ansible"
     fi
+    # Add an /etc/hosts entry for a local ipv6 address ("localhost6")
+    if ! grep -q localhost6 /etc/hosts; then
+        echo "::1         localhost6" | sudo tee -a /etc/hosts
```

Comment:
That will be incompatible with macos, where we will not want to touch it.

Interestingly on macos I see:

$ cat /etc/hosts
##
# Host Database
#
# localhost is used to configure the loopback interface
# when the system is booting.  Do not change this entry.
##
127.0.0.1   localhost
255.255.255.255 broadcasthost
::1             localhost

This makes me think that localhost works for both ipv4 and ipv6 on mac anyways.

Reply:
@ssbarnea  The reason I used `localhost6` was that `127.0.0.1`, which was used in the original code, did not work in GitHub Actions. When `localhost` was given as the mock server address, the server was listening the ipv6 address only. For enforcing the axios library used in our VS Code extension to look at the ipv6 address instead of `127.0.0.1`, I needed to have an ipv6-only entry in `/etc/hosts`.

I did some more research and found that another hostname `ip6-localhost` was available on the GitHub Action runner Ubuntu image and running server/extension with `127.0.0.1` working in macos tests.  That means:

- CI tests on (Ubuntu) Linux can use `ip6-localhost` as the hostname to resolve to the local ipv6 address
- CI tests on macos can use the ipv4 address `127.0.0.1`
- So there is no need to modify `/etc/hosts`

The new commits reflect those changes. Would you take a look? Thank you!

Reply:
This seems to work fine: https://github.com/ansible/vscode-ansible/pull/1073/commits/a32d85e9bc633cce0c7f9e4a6275d9945dcf2fe9

Reply:
The test does not seem to fail with `localhost`, but actually [the log](https://github.com/ansible/vscode-ansible/actions/runs/7731339027/job/21078753280) contains:
```
Skipping lightspeed tests because of the following error message:  connect ECONNREFUSED 127.0.0.1:3000
```
This was from [these lines](https://github.com/ansible/vscode-ansible/blob/main/test/helper.ts#L172-L176).

Reply:
Oh my bad.

Reply:
@ssbarnea Can I resolve this conversation? I would like to merge this to the main branch if it looks good.  As Ganesh pointed out in his comment below, Lightspeed tests are failing in macos test right now, but it is caused by [a known vscode bug](https://github.com/microsoft/vscode/issues/204005) and I believe it will be solved eventually (hopefully soon!).